### PR TITLE
Combined history timeline

### PR DESCRIPTION
**Pull request by bmallred**
This relates specifically to the issues listed in regards to combining the individual history sections in to one comprehensive timeline.

 - #394: Create combined review for the history section
 - #393: Create component for history timeline with gap analysis
 - #485: Branch and details for school information
 - #489: Branch and details for degree
 - #488: Summary counter/progress for schools and degrees
 - #487: Summary for school information in history collection
 - #481: Account for different "totals" for history collection based on the 18th birthday
 - #498: Make employment adjustments for self employment
 - #497: Make employment adjustments for unemployment
 - #496: Make employment adjustments for military
 - #484: Radio group for reasons leaving a job
 - #480: Adjust gap analysis logic to account for dates ending on the last day rather than the first of the month
 - #483: Delay when gaps are displayed (not immediately)
 - #482: Show in the accordion when history items are not complete

### CODE REVIEW

**Comment by bmallred** [comment]
Pending a merge from the `gap-analysis` branch

**Comment by bmallred** [comment]
Issue: #482 

incomplete

**Comment by bmallred** [comment]
Issue: #484 

reason-to-leave

**Comment by bmallred** [comment]
Issue: #488 

summary-counter

**Comment by bmallred** [comment]
Issue: #463 

Please note this is an old screenshot

**Comment by bmallred** [comment]
Issue: #487 

## School line items

school-summary

## Degree line items

degree-summary

File: src/components/Section/History/SummaryCounter/SummaryCounter.test.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      schools: () => {
+        return [
+          {
+            to: new Date(),
+            from: new Date(new Date() - 2)
+          },
+          {
+            to: new Date(new Date() - 6),
+            from: new Date(new Date() - 12)
+          }
```
Comment (mgwalker) [inline src/components/Section/History/SummaryCounter/SummaryCounter.test.jsx]: Can you add one more test case here that is outside the summary range (10 years?) to verify that there are still only two displayed?

File: src/components/Section/History/SummaryProgress/SummaryProgress.jsx [hunk 2]
```diff
@@ -14,6 +18,8 @@ export default class SummaryProgress extends ValidationElement {
       console.warn('No List() function was provided for Summary Progress')
     }
 
+    const julianMax = julian(daysAgo(today, 365 * this.total()))
```
Comment (mgwalker) [inline src/components/Section/History/SummaryProgress/SummaryProgress.jsx]: Just flagging more Julian date stuff.

File: src/components/Section/History/Education/Education.jsx [hunk 3]
```diff
@@ -0,0 +1,301 @@
+import React from 'react'
+import { i18n } from '../../../../config'
+import { ValidationElement, Branch, Collection, Comments, DateRange, DateControl, Reference, Text, RadioGroup, Radio, Help, HelpIcon, Address, Svg, Show } from '../../../Form'
+import { DiplomaItem } from './Diploma'
+import { today, daysAgo } from '../dateranges'
+
+// We need to determine how far back 3 years ago was
+const threeYearsAgo = daysAgo(today, 365 * 3)
```
Comment (bmallred) [inline src/components/Section/History/Education/Education.jsx]: I'm completely okay with using `momentjs` actually :)

Issue: #526

File: src/components/Section/History/Education/Education.jsx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+      <div className="education">
+        <Branch name="branch_school"
+                className="eapp-field-wrap"
+                value={this.state.HasAttended}
+                help="history.education.help.attendance"
+                label={i18n.t('history.education.label.attendance')}
+                onUpdate={this.updateBranchAttendance}
+                >
+        </Branch>
+        <Show when={this.state.HasAttended === 'No'}>
```
Comment (bmallred) [inline src/components/Section/History/Education/Education.jsx]: Haven't thought about it but will pose the question to the team.

File: src/components/Section/History/SummaryCounter/SummaryCounter.test.jsx (hunk 1 above)
Comment (bmallred) [inline src/components/Section/History/SummaryCounter/SummaryCounter.test.jsx]: Sure can... will track over on #527

File: src/components/Section/History/SummaryProgress/SummaryProgress.jsx (hunk 2 above)
Comment (bmallred) [inline src/components/Section/History/SummaryProgress/SummaryProgress.jsx]: We went with Julian dates for a couple of reasons:

 - It is on a linear scale
 - The days end/start at noon (may be incorrect on this) allowing for a built-in margin of error
 - Easy to compare two dates b/c it removes timezones and calendar deviations

We could go with something else but this was the beginning of making the logic at least more readable :)

**Pull request merged**